python/metrics/src/hydrotools/metrics/_validation.py

```diff
@@ -0,0 +1,68 @@
+"""
+=============================
+Method Validation for Metrics
+=============================
+Various methods and objects for validating metric inputs.
+
+Functions
+---------
+ - raise_for_non_vector
+ - raise_for_inconsistent_shapes
+
+"""
+
+import numpy as np
+import numpy.typing as npt
+from typing import List
+
+def raise_for_non_vector(
+    *arrays: List[npt.ArrayLike]
+    ) -> None:
+    """
+    Validate that all arrays are 1-dimensional.
+
+    Parameters
+    ----------
+    arrays: arbitrary number of array-like values, required
+        Arrays to check.
+
+    Raises
+    ------
+    ValueError:
+        Raises if one of the arrays is not 1-dimensional.
+    """
+    # Check each array
+    for x in arrays:
+        if len(np.array(x).shape) != 1:
+            message = "all input arrays must be 1-dimensional"
+            raise ValueError(message)
```

Comment:
In both these validation functions, the exception may be too generic.  It would be quite valuable to a developer if your messaging could indicate which array's fail validation -- or at the very least indicate what the differences are...

Reply:
I'll note that these general `_validation` functions don't need to have this functionality, you can customize your ValueError exception so you can handle that higher up:

```python
try:
   validate(...)
catch ShapeValueError:
    throw ValueError("Shape of y_true != y_pred: y_pred shape is {}, y_true shape is {}".format(y_pred.shape, y_true.shape)
```

Reply:
Updated with new `NonVectorError` and `InconsistentShapesError` derived from `Exception` that report the offending arrays.